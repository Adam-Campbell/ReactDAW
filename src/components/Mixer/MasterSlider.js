import React from 'react';
import PropTypes from 'prop-types';
import Meter from './Meter';

const MasterSlider = props => (
    <div className="mixer__master-controls-container">
        <div className="mixer__master-slider-container">
            <div className="mixer__master-volume-marker-container">
                <p className="mixer__master-volume-marker">10</p>
                <p className="mixer__master-volume-marker">0</p>
                <p className="mixer__master-volume-marker">-10</p>
                <p className="mixer__master-volume-marker">-20</p>
                <p className="mixer__master-volume-marker">-30</p>
                <p className="mixer__master-volume-marker">-40</p>
            </div>
            <input
                className="range-input range-input__mixer-master-volume"
                type="range"
                id="master-volume-slider"
                min={-40}
                max={10}
                step={0.25}
                value={props.volume}
                onChange={props.handleVolumeChange}
            ></input>
            <Meter trackId={props.trackId} isMaster />
        </div>
    </div>
); 

MasterSlider.propTypes = {
    volume: PropTypes.number.isRequired,
    handleVolumeChange: PropTypes.func.isRequired,
    trackId: PropTypes.string.isRequired
};

export default MasterSlider;