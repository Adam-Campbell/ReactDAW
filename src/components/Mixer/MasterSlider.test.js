import React from 'react';
import { shallow } from 'enzyme';
import MasterSlider from './MasterSlider';

test('renders correctly', () => {
    const component = shallow(
        <MasterSlider 
            volume={0}
            trackId="master"
            handleVolumeChange={jest.fn()}
        />
    );
    expect(component).toMatchSnapshot();
});