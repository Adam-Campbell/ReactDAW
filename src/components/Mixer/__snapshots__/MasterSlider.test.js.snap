// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="mixer__master-controls-container"
>
  <div
    className="mixer__master-slider-container"
  >
    <div
      className="mixer__master-volume-marker-container"
    >
      <p
        className="mixer__master-volume-marker"
      >
        10
      </p>
      <p
        className="mixer__master-volume-marker"
      >
        0
      </p>
      <p
        className="mixer__master-volume-marker"
      >
        -10
      </p>
      <p
        className="mixer__master-volume-marker"
      >
        -20
      </p>
      <p
        className="mixer__master-volume-marker"
      >
        -30
      </p>
      <p
        className="mixer__master-volume-marker"
      >
        -40
      </p>
    </div>
    <input
      className="range-input range-input__mixer-master-volume"
      id="master-volume-slider"
      max={10}
      min={-40}
      onChange={[MockFunction]}
      step={0.25}
      type="range"
      value={0}
    />
    <Meter
      isMaster={true}
      trackId="master"
    />
  </div>
</div>
`;
